conan/tools/meson/toolchain.py

```diff
@@ -69,21 +69,49 @@ class MesonToolchain(object):
 
     def __init__(self, conanfile, backend=None):
         self._conanfile = conanfile
-        self._backend = self._get_backend(backend)
-        self._build_type = self._conanfile.settings.get_safe("build_type")
-        self._base_compiler = self._conanfile.settings.get_safe("compiler.base") or \
-                              self._conanfile.settings.get_safe("compiler")
-        self._vscrt = self._conanfile.settings.get_safe("compiler.base.runtime") or \
-                      self._conanfile.settings.get_safe("compiler.runtime")
+        # Values are kept as Python built-ins so users can modify them more easily, and they are
```

Comment:
these were private (internal) variables, why do we need users to modify them?
in general, I expect users to modify inputs (in our world, settings and conf), instead of messing up with an internal structure of objects. our objects have public interfaces (e.g. `generate` method and few others), but directly modifying object fields doesn't sound like a good idea for me.

Reply:
The ``self.backend``, ``self.b_vscrt`` are public variables that can be changed, but are very inconveniently assigned as doubled quoted strings, not regular strings, they should be regular strings.

A design principle of all toolchains is that they should allow easy customization, like:

```python
tc = MesonToolchain(self)
tc.b_vscrt = xxxx if something else yyyy
tc.generate()
```
This is only nice if ``b_vscrt`` is a regular string

Reply:
I agree they shouldn't contain extra quotes, that's fine, I am not arguing with that at all
I just don't like to expose too many internal implementation details to the user
e.g. I didn't see anyone actually requested `b_vscrt` to be modifiable directly
we have other (and more important) actual requests for Meson (e.g. cross-building), I'd just keep `b_vscrt` and others to be fully private members for now, as they aren't currently documented anyway, and it gives us the freedom to do breaking changes
in other words, I wouldn't expose and document them right now, until it's actually requested

Reply:
about `self.backend` and `self._backend` - two are redundant and confusing, for sure
but the method `_get_backend` fully mimics `_get_generator` from `CMakeToolchain`
(backend in meson worlds is just like generator in cmake world)
it could be consistent across different toolchains, I think